### PR TITLE
NAS-127108 / 24.10 / Add narrower checks for sharing roles

### PR DESCRIPTION
**Pull request by undsoft**
Testing: test services page. You may need to create a privilege with READONLY + role to test. Or just mock.
If your new cannot login, you may need to mock `processLoginResult` in `AuthService`.

### CODE REVIEW

**Comment by undsoft** [comment]
@bvasilenko

File: src/app/directives/common/requires-roles/requires-roles.directive.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import {
   ComponentRef, Directive, HostBinding, Input, TemplateRef, ViewContainerRef,
 } from '@angular/core';
 import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { isEqual } from 'lodash';
```
Comment (bvasilenko) [inline src/app/directives/common/requires-roles/requires-roles.directive.ts:5]: Maybe it worth using `isMatch` here

**Review by bvasilenko — APPROVED** [review]
Works as expected!

File: src/app/directives/common/requires-roles/requires-roles.directive.ts (hunk 1 above)
Comment (undsoft) [reply src/app/directives/common/requires-roles/requires-roles.directive.ts:5]: Would't `isMatch` perform a partial comparison or something like it?

**Comment by undsoft** [comment]
backport

**Pull request merged**